mscore/pianoroll/pianoroll.cpp

```diff
@@ -56,118 +56,266 @@ PianorollEditor::PianorollEditor(QWidget* parent)
+      bngrpNoteLen = new QButtonGroup();
 
-      tb = addToolBar("Toolbar 2");
+      struct ToolIconData
+      {
+          QString _icon;
+          PianoRollEditTool _tool;
+          QString _tooltip;
+          bool _selected;
+      };
+      ToolIconData _iconDataTool[] = {
+            { QStringLiteral(":/data/icons/preEdit-select.svg"), PianoRollEditTool::SELECT, tr("Select Notes"), true },
+            { QStringLiteral(":/data/icons/preEdit-insertNote.svg"), PianoRollEditTool::INSERT_NOTE, tr("Insert Note"), false },
+            { QStringLiteral(":/data/icons/preEdit-appendChord.svg"), PianoRollEditTool::APPEND_NOTE, tr("Append Note to Chord"), false },
+            { QStringLiteral(":/data/icons/preEdit-cutNote.svg"), PianoRollEditTool::CUT_CHORD, tr("Cut Chord"), false },
+            { QStringLiteral(":/data/icons/preEdit-eraseNote.svg"), PianoRollEditTool::ERASE, tr("Erase Note"), false },
+            { QStringLiteral(":/data/icons/preEdit-changeLength.svg"), PianoRollEditTool::CHANGE_LENGTH, tr("Change Chord Length"), false },
+            { QStringLiteral(":/data/icons/preEdit-tie.svg"), PianoRollEditTool::TIE, tr("Toggle Tie"), false },
+            { "", PianoRollEditTool::LAST, "", false },
+            };
+
+      for (ToolIconData* p = _iconDataTool; p->_tool != PianoRollEditTool::LAST; ++p) {
+            QToolButton* bn = new QToolButton();
+            QIcon icon;
+            icon.addFile(p->_icon, QSize(), QIcon::Normal, QIcon::Off);
+            bn->setIcon(icon);
+            bn->setCheckable(true);
+            bn->setToolTip(p->_tooltip);
+            PianoRollEditTool tool = p->_tool;
+            connect(bn, &QToolButton::clicked, this, [=](){this->setEditNoteTool(tool);});
+
+            if (p->_selected)
+                  bn->setChecked(true);
+            bngrpNoteLen->addButton(bn);
+            tbTool->addWidget(bn);
+            }
+
+
+      //----
+
+      QToolBar* tbVoices = addToolBar("Toolbar Voices");
+      bngrpNoteLen = new QButtonGroup();
+
+      struct VoiceIconData
+      {
+          QString _icon;
+          int _voice;
+          QString _tooltip;
+          bool _selected;
+      };
+      VoiceIconData _iconDataVoice[] = {
+            { QStringLiteral(":/data/icons/voice-1.svg"), 0, tr("Voice 1"), true },
+            { QStringLiteral(":/data/icons/voice-2.svg"), 1, tr("Voice 2"), false },
+            { QStringLiteral(":/data/icons/voice-3.svg"), 2, tr("Voice 3"), false },
+            { QStringLiteral(":/data/icons/voice-4.svg"), 3, tr("Voice 4"), false },
+            { "", -1, "", false },
+            };
+
+      for (VoiceIconData* p = _iconDataVoice; p->_voice != -1; ++p) {
+            QToolButton* bn = new QToolButton();
+            QIcon icon;
+            icon.addFile(p->_icon, QSize(), QIcon::Normal, QIcon::Off);
+            bn->setIcon(icon);
+            bn->setCheckable(true);
+            bn->setToolTip(p->_tooltip);
```

Comment:
here it might be possible to 'steal' the translation using `qApp->translate("action", p->_toolTip)` and drop the `tr(...)` above in the initialization

Reply:
I think I will keep the strings separate to avoid any clashes.  It's only four strings and it would be best to not have it depend on a completely different editor window.